src/editor/blocks/posts-inserter/index.js

```diff
@@ -38,12 +45,51 @@ const PostsInserterBlock = ( {
 	setInsertedPostsIds,
 	removeBlock,
 } ) => {
+	const [ ready, setReady ] = useState( ! attributes.displayFeaturedImage );
+
 	// Stringify added to minimize flicker.
 	const templateBlocks = useMemo( () => getTemplateBlocks( postList, attributes ), [
 		JSON.stringify( postList ),
 		attributes,
 	] );
 
+	useEffect(() => {
+		const { isDisplayingSpecificPosts, specificPosts } = attributes;
+
+		// No spinner if we're not dealing with images.
+		if ( ! attributes.displayFeaturedImage ) {
+			return setReady( true );
+		}
+
+		// No spinner if we're in the middle of selecting a specific post.
+		if ( isDisplayingSpecificPosts && 0 === specificPosts.length ) {
+			return setReady( true );
+		}
+
+		// Reset ready state.
+		setReady( false );
+
+		// If we have a post to show, check for featured image blocks.
+		if ( 0 < postList.length ) {
+			// Find all the featured images.
+			const images = [];
+			postList.map( post => post.featured_media && images.push( post.featured_media ) );
+
+			// If no posts have featured media, skip loading state.
+			if ( 0 === images.length ) {
+				return setReady( true );
+			}
+
+			// Wait for image blocks to be added to the BlockPreview.
+			const imageBlocks = templateBlocks.filter( block => 'core/image' === block.name );
```

Comment:
This will handle only image blocks which are top-level blocks. When testing with the "Daily/Weekly" template the feature did not work, because the image blocks there are nested in column blocks.

Reply:
@adekbadek Thanks for pointing out—didn't realize this could contain images, too. Will revisit to see if I can capture those as well.

Reply:
This should handle image blocks no matter how deeply nested—what do you think? d846beb